MidasCivil_Adapter/CRUD/Create/Loads/BarDifferentialTemperatureLoad.cs

```diff
@@ -0,0 +1,79 @@
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using BH.oM.Adapters.MidasCivil;
+using BH.Engine.Adapter;
+using BH.oM.Structure.Loads;
+using BH.oM.Structure.Elements;
+using System.Collections.Generic;
+using System.IO;
+using BH.oM.Structure.SectionProperties;
+
+namespace BH.Adapter.MidasCivil
+{
+    public partial class MidasCivilAdapter
+    {
+        /***************************************************/
+        /**** Private Methods                           ****/
+        /***************************************************/
+
+        private bool CreateCollection(IEnumerable<BarDifferentialTemperatureLoad> barDifferentialTemperatureLoads)
+        {
+            string loadGroupPath = CreateSectionFile("LOAD-GROUP");
+            foreach (BarDifferentialTemperatureLoad barDifferentialTemperatureLoad in barDifferentialTemperatureLoads)
+            {
+                if (barDifferentialTemperatureLoad.TemperatureProfile.Keys.Count > 21)
+                {
+                    Engine.Reflection.Compute.RecordWarning("There is a maximium of 20 layers in the temperature profile");
+                    return true;
+                }
+                List<string> midasTemperatureLoads = new List<string>();
+                string barLoadPath = CreateSectionFile(barDifferentialTemperatureLoad.Loadcase.Name + "\\BSTEMPER");
+                string midasLoadGroup = Adapters.MidasCivil.Convert.FromLoadGroup(barDifferentialTemperatureLoad);
+                var groupedBars = barDifferentialTemperatureLoad.Objects.Elements.GroupBy(x => x.SectionProperty);
+                ISectionProperty sectionPropertyCheck = groupedBars.First().Key;
+                foreach (var barGroup in groupedBars)
+                {
+                    string ids = "";
+                    ISectionProperty sectionProperty = barGroup.Key;
+                    if (sectionProperty == null)
+                    {
+                        Engine.Reflection.Compute.RecordWarning("Section Property is required for inputting differential temperature load");
+                        return true;
+                    }
```

Comment:
This `null` check needs to come before you group the `Bars` by `SectionProperty` (otherwise it throws an exception).

```
if(barDifferentialTemperatureLoad.Objects.Elements.Any(x => x.SectionProperty == null)
{
	Engine.Reflection.Compute.RecordWarning("Section Property is required for inputting differential temperature load");
        return true;
}
```